test/BrickFarming-Unit.test.waffle.chai.ethers.js

```diff
@@ -0,0 +1,314 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+const { BigNumber } = ethers;
+const { ether, gWei, wei } = require("./utils/unitUtils");
+const { ZERO } = require("./utils/constants");
+const { getLatestBlockTimestamp } = require("./utils/helpers");
+const { advanceTimeAndBlock } = require("./utils/timeUtils");
+
+describe("BrickFarming", function () {
+    let owner;
+    let rewardToken;
+    let stakedToken;
+    let treasury;
+    let farming;
+
+    let tester1;
+    let tester2;
+    let tester3;
+
+    const lockupDuration = BigNumber.from(3 * 24 * 3600); // 3 days
+
+    before(async function () {
+        const signers = await ethers.getSigners();
+
+        owner = signers[0];
+        tester1 = signers[1];
+        tester2 = signers[2];
+        tester3 = signers[3];
+        treasury = signers[4].address;
+
+        RewardTokenContract = await ethers.getContractFactory("MockBrickToken");
+        rewardToken = await RewardTokenContract.connect(owner).deploy();
+
+        BrickFarmingFactory = await ethers.getContractFactory("BrickFarming");
+        farming = await BrickFarmingFactory.connect(owner).deploy(
+            rewardToken.address,
+            treasury
+        );
+
+        StakedTokenContract = await ethers.getContractFactory(
+            "MockStakedToken"
+        );
+        stakedToken = await StakedTokenContract.connect(owner).deploy();
+
+        // transfer lptokens to users
+        await stakedToken.transfer(tester1.address, ether(50));
+        await stakedToken.transfer(tester2.address, ether(50));
+        await stakedToken.transfer(tester3.address, ether(50));
+
+        // approve
+        await rewardToken.approve(farming.address, ether(10000));
+        await stakedToken
+            .connect(tester1)
+            .approve(farming.address, ether(10000));
```

Comment:
@bejaxer 
why is user balance 50 and the approval 10000?

Reply:
> @bejaxer why is user balance 50 and the approval 10000?

approve amount is demo one - 10000 > 50 so we can think it's an infinite-approve :)

Reply:
can we do `type(uint256).max` in JS then?